### PR TITLE
Feature/ap 4

### PR DESCRIPTION
**Pull request by rdhawladar**
User login

### CODE REVIEW

File: internal/app/model/entities.go [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+
 type User struct {
 	ID          int    `json:"id"`
 	PhoneNumber string `json:"phone_number" db:"phone_number"`
 	Password    string `json:"password"`
 	CategoryId  int    `json:"category_id" db:"category_id"`
 }
+
+// PhoneValidation Validate phone numbers with regexp
+func PhoneValidation(phoneNumber string) bool {
```
Comment (shuza) [inline internal/app/model/entities.go:15]: bind this method with user struct and do all validation in one method

File: internal/app/server/handler.go [hunk 2]
```diff
@@ -1 +1,39 @@
...
+		return
+	}
+
+	token, err := s.authService.Encode(user)
+	if err != nil {
+		log.Error().Err(err).Msgf("[login] failed to generate jwt token Error: %v", err)
+		ErrInternalServerResponse(w, "failed to generate jwt token", err)
+		return
+	}
+	SuccessResponse(w, http.StatusOK, token)
```
Comment (shuza) [inline internal/app/server/handler.go:38]: return all user data with token

File: internal/app/server/halder_test.go [hunk 3]
```diff
@@ -1 +1,117 @@
...
+		PhoneNumber: "123456",
+		CategoryId:  1,
+	}
+
+	testCases := []struct {
+		desc          string
+		payload       string
+		mockUserSvc   func() *mocks.MockUserService
+		mockAuthSvc   func() *mocks.MockAuthService
+		expStatusCode int
```
Comment (shuza) [inline internal/app/server/halder_test.go:33]: also need to check response data

**Review by shuza — CHANGES_REQUESTED** [review]
* need to check response data
* need to return user data with token
* do all login validation in one method

File: internal/app/server/handler.go (hunk 2 above)
Comment (rdhawladar) [reply internal/app/server/handler.go]: Returned all user data.

File: internal/app/model/entities.go (hunk 1 above)
Comment (rdhawladar) [reply internal/app/model/entities.go]: Added single validation method for login.

**Review by shuza — APPROVED** [review]
lgtm

**Pull request merged**